lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1075,13 +1075,13 @@ def test_get_html_hls(self, get_urls_for_profiles):
         }
 
         self.initialize_block(data=video_xml)
-        context = self.item_descriptor.render(STUDENT_VIEW).content
+        context = get_context_dict_from_string(self.item_descriptor.render(STUDENT_VIEW).content)
 
-        self.assertIn("'download_video_link': 'https://mp4.com/dm.mp4'", context)
-        self.assertIn('"streams": "1.00:https://yt.com/?v=v0TFmdO4ZP0"', context)
+        self.assertEqual('https://mp4.com/dm.mp4', context['download_video_link'])
```

Comment:
This was a flaky test, flakiness was caused by [unordered dict](https://github.com/edx-solutions/edx-platform/blob/rebase-juniper/common/lib/xmodule/xmodule/video_module/video_module.py#L460). Because this an unordered dict when [pprint prints](https://github.com/edx-solutions/edx-platform/blob/rebase-juniper/common/lib/xmodule/xmodule/tests/__init__.py#L178) it, It adds `\n` based on the order of attributes due to which it sometime fails. Converting string content to dict and then checking values fixes the problem.